ui/jobs/jobs.ts

```diff
@@ -396,41 +482,49 @@ class Bars {
     return textDiv;
   }
 
-  addProcBox(options) {
+  addProcBox(options: {
+    id?: string;
+    fgColor?: string;
+    threshold?: number;
+    scale?: number;
+  }): TimerBox {
     const id = this.job.toLowerCase() + '-procs';
 
     let container = document.getElementById(id);
     if (!container) {
       container = document.createElement('div');
       container.id = id;
-      document.getElementById('bars').appendChild(container);
+      document.getElementById('bars')?.appendChild(container);
       container.classList.add('proc-box');
     }
 
-    const timerBox = document.createElement('timer-box');
+    const timerBox = document.createElement('timer-box') as unknown as TimerBox;
     container.appendChild(timerBox);
     timerBox.stylefill = 'empty';
     if (options.fgColor)
       timerBox.fg = computeBackgroundColorFrom(timerBox, options.fgColor);
     timerBox.bg = 'black';
     timerBox.toward = 'bottom';
-    timerBox.threshold = options.threshold ? options.threshold : 0;
+    timerBox.threshold = `${options.threshold ? options.threshold : 0}`;
     timerBox.hideafter = '';
     timerBox.roundupthreshold = false;
-    timerBox.valuescale = options.scale ? options.scale : 1;
+    timerBox.valuescale = `${options.scale ? options.scale : 1}`;
     if (options.id) {
       timerBox.id = options.id;
       timerBox.classList.add('timer-box');
     }
     return timerBox;
   }
 
-  addTimerBar(options) {
+  addTimerBar(options: {
+    id: string;
+    fgColor: string;
+  }): TimerBar {
     const container = this.addJobBarContainer();
 
     const timerDiv = document.createElement('div');
     timerDiv.id = options.id;
-    const timer = document.createElement('timer-bar');
+    const timer = document.createElement('timer-bar') as TimerBar;
```

Comment:
I was talking to @panicstevenson about this, but I wonder if we need a static `create` function so we could hide this `as` and `timer-bar` implicit name, e.g. like a `create(): TimerBar` function on `TimerBar`.  I don't think that should be done in this patch (maybe before? or after?), but it's something on my mind.

Reply:
Maybe a "Factory class" is better?
i.e. something like
```typescript
class CustomElementFactory {
  static create(s: "timer-bar"): TimerBar;
  static create(s: "timer-icon"): TimerIcon;
  static create(s: string): HTMLElement {
    // return custom element based on the parameter.
  }
}
```